### ISSUE TITLE
Put cursor at the end of text in EditText of AnkiDroid directory

### ISSUE THREAD

**Issue by Abanoub-Asaad** [ankidroid/Anki-Android#8336]
When I open 
`Settings > Advanced > AnkiDroid directory`
The cursor is at the start of text in EditText. It should be at the end of the text for **better experience**.

**Comment by david-allison-1** [comment]
I'd like this, would you like to work on it?

**Comment by Abanoub-Asaad** [comment]
> I'd like this, would you like to work on it?

Yes, I want to work on it.

**Comment by david-allison-1** [comment]
To note: there are a few more complex feature requests to turn this dialog into a file picker with a "restore default".

I'm not sure how long these will take to get to get through (or if anyone will work on them), I'd be happy with this on its own, but thought you'd want to be aware.

**Comment by Abanoub-Asaad** [comment]
> To note: there are a few more complex feature requests to turn this dialog into a file picker with a "restore default".
> 
> I'm not sure how long these will take to get to get through (or if anyone will work on them), I'd be happy with this on its own, but thought you'd want to be aware in case it's something you'd rather work on

Thanks for your notes. I'll be aware of them.

**Comment by mikehardy** [comment]
Just a note - if this is turned in to a file picker then you can't assign "does not exist yet" directories to it, to be specific on first start /sdcard/AnkiDroid does not exist, and if I install a parallel build and I want it to store things in /sdcard/AnkiDroid.A it won't exist yet, I won't be able to pick it.

I think this whole thing "goes away" actually and we do not allow it to be configurable anymore when we stop using external storage.

**Comment by Abanoub-Asaad** [comment]
> Just a note - if this is turned in to a file picker then you can't assign "does not exist yet" directories to it, to be specific on first start /sdcard/AnkiDroid does not exist, and if I install a parallel build and I want it to store things in /sdcard/AnkiDroid.A it won't exist yet, I won't be able to pick it.
> 
> I think this whole thing "goes away" actually and we do not allow it to be configurable anymore when we stop using external storage.

Did you mean that I "As a user" can't edit the path to a folder in external storage and only all I can do is to edit the path to an internal storage folder?

**Comment by Abanoub-Asaad** [comment]
Can you tell me Where the Settings' related classes in the code because I can't find them?

**Comment by david-allison-1** [comment]
https://github.com/ankidroid/Anki-Android/blob/master/AnkiDroid/src/main/res/xml/preferences_advanced.xml#L29-L33

https://github.com/ankidroid/Anki-Android/blob/master/AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java

**Comment by Abanoub-Asaad** [comment]
> To note: there are a few more complex feature requests to turn this dialog into a file picker with a "restore default".
> 
> I'm not sure how long these will take to get to get through (or if anyone will work on them), I'd be happy with this on its own, but thought you'd want to be aware in case it's something you'd rather work on

**Firstly**, sorry, I don't understand what you mean when you said "restore default"

**Secondly**, I checked most of the dialogs in the code base, but I couldn't find the code of the target dialog which is used to edit the path. I'm trying to find this dialog to access it and its related methods like `OnPreferenceClickListener`.

**Comment by david-allison-1** [comment]
> > To note: there are a few more complex feature requests to turn this dialog into a file picker with a "restore default".
> > I'm not sure how long these will take to get to get through (or if anyone will work on them), I'd be happy with this on its own, but thought you'd want to be aware in case it's something you'd rather work on
> 
> **Firstly**, sorry, I don't understand what you mean when you said "restore default"

Restore to the default value: 

Some users have this value modified due to the backup process between devices

https://github.com/ankidroid/Anki-Android/blob/69dba64459dc466daad2d81012c55b9d66b721bb/AnkiDroid/src/main/java/com/ichi2/anki/CollectionHelper.java#L228-L237

> **Secondly**, I checked most of the dialogs in the code base, but I couldn't find the code of the target dialog which is used to edit the path. I'm trying to find this dialog to access it and its related methods like `OnPreferenceClickListener`.

Preferences reads `EditTextPreference` and converts it into a dialog.

https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/preference/EditTextPreference.java#141

**Comment by Abanoub-Asaad** [comment]
> https://github.com/ankidroid/Anki-Android/blob/3dd1e8506425add2552bf65ada1664b76eec321c/AnkiDroid/src/main/res/xml/preferences_advanced.xml#L29-L33
> 
> https://github.com/ankidroid/Anki-Android/blob/master/AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java

<br>

When I'm trying to implement the target feature in this method `setOnPreferenceClickListener`, I'm facing some problems.

**Comment by david-allison-1** [comment]
Intuition tells me it'll be easier to implement this via inheriting from `EditTextPreference`

**Comment by Abanoub-Asaad** [comment]
Actually, this What I tried to do
```java
CollectionPathPreference.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
                    @Override
                    public boolean onPreferenceClick(Preference preference) {
                        collectionPathPreference.getEditText().setSelection(collectionPathPreference.getText().length());
                        return false;
                    }
                });
```

**Comment by Abanoub-Asaad** [comment]
> Intuition tells me it'll be easier to implement this via inheriting from `EditTextPreference`

When I'm trying to do so, I'm facing a lot of problems "errors" because of the deprecation of these classes.
Why are you still using `Preferences` and now it's deprecated?

**Comment by david-allison-1** [comment]
https://github.com/ankidroid/Anki-Android/issues/5019 . You can work around the issue for now by fully qualifying the type names and removing the imports.

For example: https://github.com/ankidroid/Anki-Android/blob/e034b3851f1b7de0f5864d3c7f830e19de4e5d87/AnkiDroid/src/main/java/com/ichi2/preferences/StepsPreference.java#L32

**Comment by david-allison-1** [comment]
This should be one line of code after #9397.

On second thought, it will conflict with the scoped storage change, so will leave it until that settles, but I'd like to see this if it's simple.

**Comment by milin831** [comment]
yes I want to work on it

**Comment by milin831** [comment]
yes I want to work on it
if these issues have not solved yet

**Comment by david-allison-1** [comment]
sounds good to me

**Comment by david-allison-1** [comment]
I'm fine to see this one go stale: 

low priority, and will interact badly with scoped storage changes
